packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -61,22 +67,35 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
     }
   }, [response]);
 
-  const setItems = useCallback((newResponseItems: QuestionnaireResponseItem | QuestionnaireResponseItem[]): void => {
-    setResponse((prevResponse) => {
-      const currentItems = prevResponse?.item ?? [];
-      const mergedItems = mergeItems(
-        currentItems,
-        Array.isArray(newResponseItems) ? newResponseItems : [newResponseItems]
-      );
-
-      const newResponse: QuestionnaireResponse = {
-        resourceType: 'QuestionnaireResponse',
-        status: 'in-progress',
-        item: mergedItems,
-      };
-      return newResponse;
-    });
-  }, []);
+  const setItems = useCallback(
+    (newResponseItems: QuestionnaireResponseItem | QuestionnaireResponseItem[]): void => {
+      setResponse((prevResponse) => {
+        const currentItems = prevResponse?.item ?? [];
+        const mergedItems = mergeItems(
+          currentItems,
+          Array.isArray(newResponseItems) ? newResponseItems : [newResponseItems]
+        );
+
+        const tempResponse: QuestionnaireResponse = {
+          resourceType: 'QuestionnaireResponse',
+          status: 'in-progress',
+          item: mergedItems,
+        };
+
+        const updatedItems = evaluateCalculatedExpressionsInQuestionnaire(questionnaire?.item ?? [], tempResponse);
+        const mergedItemsWithUpdates = mergeUpdatedItems(mergedItems, updatedItems);
+
+        const newResponse: QuestionnaireResponse = {
+          resourceType: 'QuestionnaireResponse',
+          status: 'in-progress',
+          item: mergedItemsWithUpdates,
+        };
+
+        return newResponse;
+      });
+    },
+    [questionnaire]
```

Comment:
I think we talked about this before right @mattlong ? Do you remember anything about the conversation we had about memoizing based on the passed in questionnaire?  Not sure if it's relevant here but wanted to flag it

Reply:
Adding questionnaire as dependency because we need to compare against responses, but given the use case here `questionnaire` wouldn't change causing the hook to re execute, right ?

Reply:
There is a case where the questionnaire's identity is different on every rerender of the top level component:

```ts
<QuestionnaireForm questionnaire={{ resourceType: 'Questionnaire', id: '123'  }} />
```

What will happen if that `useEffect` triggers on every rerender of the parent component?